src/c-opentimelineio/tests/OTIOCompositionTests.cpp

```diff
@@ -98,4 +91,163 @@ TEST_F(OTIOCompositionTests, ConstructorTest)
     retainerItEnd = NULL;
     ComposableVectorIterator_destroy(vectorIt);
     vectorIt = NULL;
+}
+
+TEST_F(OTIOCompositionTests, EqualityTest)
+{
+    Composition* co0  = Composition_create(NULL, NULL, NULL, NULL, NULL);
+    Composition* co00 = Composition_create(NULL, NULL, NULL, NULL, NULL);
+    EXPECT_TRUE(SerializableObject_is_equivalent_to(
+        (SerializableObject*) co0, (SerializableObject*) co00));
+
+    Item*             a   = Item_create("A", NULL, NULL, NULL, NULL);
+    Item*             b   = Item_create("B", NULL, NULL, NULL, NULL);
+    Item*             c   = Item_create("C", NULL, NULL, NULL, NULL);
+    Composition*      co1 = Composition_create(NULL, NULL, NULL, NULL, NULL);
+    ComposableVector* composableVector = ComposableVector_create();
+    ComposableVector_push_back(composableVector, (Composable*) a);
+    ComposableVector_push_back(composableVector, (Composable*) b);
+    ComposableVector_push_back(composableVector, (Composable*) c);
+    OTIOErrorStatus* errorStatus = OTIOErrorStatus_create();
+
+    bool resultOK =
+        Composition_set_children(co1, composableVector, errorStatus);
+    ASSERT_TRUE(resultOK);
+    ComposableVector_destroy(composableVector);
+    composableVector = NULL;
+
+    Item*        x   = Item_create("X", NULL, NULL, NULL, NULL);
+    Item*        y   = Item_create("Y", NULL, NULL, NULL, NULL);
+    Item*        z   = Item_create("Z", NULL, NULL, NULL, NULL);
+    Composition* co2 = Composition_create(NULL, NULL, NULL, NULL, NULL);
+    composableVector = ComposableVector_create();
+    ComposableVector_push_back(composableVector, (Composable*) x);
+    ComposableVector_push_back(composableVector, (Composable*) y);
+    ComposableVector_push_back(composableVector, (Composable*) z);
+    resultOK = Composition_set_children(co2, composableVector, errorStatus);
+    ASSERT_TRUE(resultOK);
+    ComposableVector_destroy(composableVector);
+    composableVector = NULL;
+
+    Item*        a2  = Item_create("A", NULL, NULL, NULL, NULL);
+    Item*        b2  = Item_create("B", NULL, NULL, NULL, NULL);
+    Item*        c2  = Item_create("C", NULL, NULL, NULL, NULL);
+    Composition* co3 = Composition_create(NULL, NULL, NULL, NULL, NULL);
+    composableVector = ComposableVector_create();
+    ComposableVector_push_back(composableVector, (Composable*) a2);
+    ComposableVector_push_back(composableVector, (Composable*) b2);
+    ComposableVector_push_back(composableVector, (Composable*) c2);
+    resultOK = Composition_set_children(co3, composableVector, errorStatus);
+    ASSERT_TRUE(resultOK);
+    ComposableVector_destroy(composableVector);
+    composableVector = NULL;
+
+    EXPECT_FALSE(SerializableObject_is_equivalent_to(
+        (SerializableObject*) co1, (SerializableObject*) co2));
+    /*EXPECT_TRUE(SerializableObject_is_equivalent_to( //TODO Fix Segfault
```

Comment:
There are some more segfaults like this that need to be fixed, but I have no idea at the moment why they're happening.